docs/blog/_posts/2019-03-04-13th-dotty-milestone-release.md

```diff
@@ -0,0 +1,439 @@
+examples above while they demonstrate completely different things, a typeclass
+instance is an implicit object or val if unconditional and an implicit def with
+implicit parameters if conditional; arguably all of them are surprisingly
+similar (syntactically). Another consideration is that the `implicit` keyword
+annotates a whole parameter section instead of a single parameter, and passing
+an argument to an implicit parameter looks like a regular application. This is
+problematic because it can create confusion regarding what parameter gets passed
+in a call. Last but not least, sometimes implicit parameters are merely
+propagated in nested function calls and not used at all, so names of implicit
+parameters are not always necessary.
+
+Consequently, we introduce two new language features:
+
+1. _implied instance definitions_ designated syntactically by the scheme `implied ... for` and,
+2. _inferable parameters_ designated by the keyword `given`.
+
+In the code below we demonstrate both of them. This code defines a trait `Ord`
+and two `implied` instance definitions. `IntOrd` defines an `implied` instance
+for the type `Ord[Int]` whereas `ListOrd[T]` defines implied instances of
+`Ord[List[T]]` for all types `T` that come with an implied `Ord[T]` instance
+themselves. The `given` clause in `ListOrd` defines an _inferable parameter_.
+
+```scala
+trait Ord[T] {
+  def compare(x: T, y: T): Int
+  def (x: T) < (y: T) = compare(x, y) < 0
+  def (x: T) > (y: T) = compare(x, y) > 0
+}
+
+implied IntOrd for Ord[Int] {
+  def compare(x: Int, y: Int) =
+    if (x < y) -1 else if (x > y) +1 else 0
+}
+
+implied ListOrd[T] given (ord: Ord[T]) for Ord[List[T]] {
+  def compare(xs: List[T], ys: List[T]): Int = (xs, ys) match {
+    case (Nil, Nil) => 0
+    case (Nil, _) => -1
+    case (_, Nil) => +1
+    case (x :: xs1, y :: ys1) =>
+      val fst = ord.compare(x, y)
+      if (fst != 0) fst else xs1.compareTo(ys1)
+  }
+}
+```
+
+A `given` clause can also designate an inferable parameter for functions:
+
+```scala
+def max[T](x: T, y: T) given (ord: Ord[T]): T =
+  if (ord.compare(x, y) < 1) y else x
+```
+
+With this scheme all invocations of the `max` function below are equally valid:
+```scala
+max(2, 3) given IntOrd
+max(List(1, 2, 3), Nil)
+max(2, 3)
+```
+
+We introduce _Anonymous Implied Instances_ which are used when we do not need a name for an implied
+instance:
+
+```scala
+implied for Ord[Int] { ... }
+```
+
+For convenience, we also introduce _Implied Alias Instances_. They offer aliases
+for implied instances. For example, the line below offers an alias with the name
+`ctx` (could also be anonymous if name can be omitted). Each time an implied
+instance of `ExecutionContext` is demanded the right-hand side is returned.
+
+```scala
+implied ctx for ExecutionContext = currentThreadPool().context
+```
+
+For symmetry, we define our well-known `implicitly` from `Predef` in terms of

```

Comment:
Does it really matter how `implicitly` is implemented?

```suggestion
We have also added a synonym to `implicitly`, which is often more natural to spell out in user code
```

(Apparently we'd keep both `the` and `implicitly` https://github.com/lampepfl/dotty/pull/5893#issuecomment-463634860 )